### PR TITLE
build: only produce binary for linux

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

We now build rook to build only for linux platforms and not darwin nor
windows anymore.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: build/makelib/common.mk [hunk 1]
```diff
@@ -51,19 +47,38 @@ ifeq ($(PLATFORM),linux_arm64)
 CROSS_TRIPLE = aarch64-linux-gnu
 endif
 ifeq ($(PLATFORM),darwin_amd64)
+# force the build of a linux binary when running on MacOS
+GOHOSTOS=linux
 CROSS_TRIPLE=x86_64-apple-darwin15
 endif
 ifeq ($(PLATFORM),windows_amd64)
+# force the build of a linux binary when running on Windows
+GOHOSTOS=linux
```
Comment (travisn) [inline build/makelib/common.mk:56]: Should `GOHOSTOS=linux` be set always, for example on line 60 next to the other vars?

File: build/makelib/golang.mk [hunk 2]
```diff
@@ -152,7 +152,8 @@ go.lint: $(GOLINT)
 .PHONY: go.vet
 go.vet:
 	@echo === go vet
-	@CGO_ENABLED=0 $(GOHOST) vet $(GO_COMMON_FLAGS) $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES)
+	@echo CGO_ENABLED=$(CGO_ENABLED_VALUE) $(GOHOST) vet $(GO_COMMON_FLAGS) $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES)
```
Comment (travisn) [inline build/makelib/golang.mk:155]: Should this echo be removed, or useful to keep?

File: Makefile [hunk 3]
```diff
@@ -76,6 +78,9 @@ GO_PROJECT=github.com/rook/rook
 # inject the version number into the golang version package using the -X linker flag
 LDFLAGS += -X $(GO_PROJECT)/pkg/version.Version=$(VERSION)
 
+# CGO_ENABLED value
+CGO_ENABLED_VALUE=0
```
Comment (travisn) [inline Makefile:82]: Was there a need to factor this out to be configurable? Its value is still always 0, right?

File: build/makelib/golang.mk (hunk 2 above)
Comment (leseb) [reply build/makelib/golang.mk:155]: I've it to be useful to see the args we use when running.

File: build/makelib/common.mk (hunk 1 above)
Comment (leseb) [reply build/makelib/common.mk:56]: I suppose this should work yes.

File: build/makelib/golang.mk (hunk 2 above)
Comment (BlaineEXE) [reply build/makelib/golang.mk:155]: Alternately, we can just remove the `@` before the line below, and then we don't need this echo. ;)

File: build/makelib/golang.mk [hunk 4]
```diff
@@ -152,7 +152,8 @@ go.lint: $(GOLINT)
...
+	@CGO_ENABLED=$(CGO_ENABLED_VALUE) $(GOHOST) vet $(GO_COMMON_FLAGS) $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES)
```
Comment (BlaineEXE) [inline build/makelib/golang.mk:156]: This does the same thing.
```suggestion
	CGO_ENABLED=$(CGO_ENABLED_VALUE) $(GOHOST) vet $(GO_COMMON_FLAGS) $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES)
```

File: build/makelib/golang.mk (hunk 2 above)
Comment (leseb) [reply build/makelib/golang.mk:155]: Rah I knew we had discussed that earlier! Thanks for the reminder.

**Review by travisn — APPROVED** [review]

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**